rust/sql/src/postgres/reader.rs

```diff
@@ -0,0 +1,894 @@
+        // &Type::NUMERIC => Some(DataType::Float64),
+        //        &OID => None,
+        //        &JSON => None,
+        &Type::FLOAT4 => Some(DataType::Float32),
+        &Type::FLOAT8 => Some(DataType::Float64),
+        //        &ABSTIME => None,
+        //        &RELTIME => None,
+        //        &TINTERVAL => None,
+        //        &MONEY => None,
+        &Type::BOOL_ARRAY => Some(DataType::List(Box::new(Field::new(
+            "list",
+            DataType::Boolean,
+            true,
+        )))),
+        &Type::BYTEA_ARRAY | &Type::CHAR_ARRAY | &Type::NAME_ARRAY => Some(
+            DataType::List(Box::new(Field::new("list", DataType::Utf8, true))),
+        ),
+        //        &INT2_ARRAY => None,
+        //        &INT2_VECTOR => None,
+        //        &INT2_VECTOR_ARRAY => None,
+        //        &INT4_ARRAY => None,
+        //        &TEXT_ARRAY => None,
+        //        &INT8_ARRAY => None,
+        //        &FLOAT4_ARRAY => None,
+        //        &FLOAT8_ARRAY => None,
+        //        &ABSTIME_ARRAY => None,
+        //        &RELTIME_ARRAY => None,
+        //        &TINTERVAL_ARRAY => None,
+        &Type::DATE => Some(DataType::Date32(DateUnit::Day)),
+        &Type::TIME => Some(DataType::Time64(TimeUnit::Microsecond)),
+        &Type::INTERVAL => Some(DataType::Interval(IntervalUnit::DayTime)),
+        &Type::TIMESTAMP | &Type::TIMESTAMPTZ => {
+            Some(DataType::Timestamp(TimeUnit::Microsecond, None))
+        }
+        //        &TIMESTAMP_ARRAY => None,
+        &Type::DATE_ARRAY => Some(DataType::List(Box::new(Field::new(
+            "item",
+            DataType::Date32(DateUnit::Day),
+            true,
+        )))),
+        //        &TIME_ARRAY => None,
+        //        &TIMESTAMPTZ => None,
+        //        &TIMESTAMPTZ_ARRAY => None,
+        //        &INTERVAL => None,
+        //        &INTERVAL_ARRAY => None,
+        //        &NUMERIC_ARRAY => None,
+        //        &TIMETZ => None,
+        //        &BIT => None,
+        //        &BIT_ARRAY => None,
+        //        &VARBIT => None,
+        //        &NUMERIC => None,
+        &Type::UUID => Some(DataType::FixedSizeBinary(16)),
+        t => panic!("Postgres type {:?} not supported", t),
+    }
+}
+
+fn read_table_by_rows(
+    connection_string: &str,
+    table_name: &str,
+    _limit: usize,
+    batch_size: usize,
+) -> Result<Vec<RecordBatch>> {
+    // create connection
+    let mut client = Client::connect(connection_string, NoTls).unwrap();
+    let results = client
+        .query(format!("SELECT * FROM {}", table_name).as_str(), &[])
+        .unwrap();
+    if results.is_empty() {
+        return Ok(vec![]);
+    }
+
+    // TODO: replace with metadata version of call
+    let schema = row_to_schema(results.get(0).unwrap()).unwrap();
+    let field_len = schema.fields().len();
+    let mut builder = StructBuilder::from_fields(schema.fields().clone(), batch_size);
```

Comment:
From @jorgecarleitao 's work, I'd expect the builder approach to be slow, so we can find better approaches that also minimise allocations.